rapidhash.h

```diff
@@ -237,20 +264,135 @@
    return (((v >> 24) & 0xff)| ((v >>  8) & 0xff00)| ((v <<  8) & 0xff0000)| ((v << 24) & 0xff000000));
  }
  #endif
- 
- /*
-  *  rapidhash main function.
-  *
-  *  @param key     Buffer to be hashed.
-  *  @param len     @key length, in bytes.
-  *  @param seed    64-bit seed used to alter the hash result predictably.
-  *  @param secret  Triplet of 64-bit secrets used to alter hash result predictably.
-  *
-  *  Returns a 64-bit hash.
-  */
+
+/**
+ * When using sequential seeds, premix them to avoid similarity between seeds.
+ *
+ * @ param seed 64-bit seed.
+ *
+ * @ return 64-bit premixed seed.
+ */
+RAPIDHASH_INLINE_CONSTEXPR uint64_t rapid_preseed(const uint64_t seed) RAPIDHASH_NOEXCEPT {
+  return rapid_mix(seed ^ rapid_secret[2], rapid_secret[1]);
+}
+
+/**
+ * Internal rapidhash cold path for inputs > 288
+ *
+ * Force no inlining to avoid the hot path on small inputs having clobbered registers that need restoring. Also makes
+ * the hot path easier to inline.
+ */
+RAPIDHASH_CONSTEXPR RAPIDHASH_NO_INLINE uint64_t rapidhash_internal_cold(const uint8_t *p, size_t len, uint64_t seed, const uint64_t* secret) RAPIDHASH_NOEXCEPT {
+  size_t i = len;
+  uint64_t see1 = seed, see2 = seed;
+  uint64_t see3 = seed, see4 = seed;
+  uint64_t see5 = seed, see6 = seed;
+#ifdef RAPIDHASH_UNROLLED

```

Comment:
We could remove these macros with V3, as we have no-inline attributes on the cold path, so these instructions will only be included once in the final binary.